src/arrayMethodSort.js

```diff
@@ -4,9 +4,31 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = defaultSort) {
+    let temp;
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (compareFunction(this[i], this[j]) > 0) {
+          temp = this[i];
+          this[i] = this[j];
+          this[j] = temp;
+        }
+      }
+    }
+
+    return this;
   };
 }
 
+function defaultSort(a, b) {
+  if (String(a) > String(b)) {
+    return 1;
+  } else if (String(a) < String(b)) {
+    return -1;
+  } else {
+    return 0;
+  }
```

Comment:
no need for `else` nesting when `if` has a `return` statement
```suggestion
  if (String(a) > String(b)) {
    return 1;
  }
  
  if (String(a) < String(b)) {
    return -1;
  }
  
  return 0;
```